### PR TITLE
[DotNet] Add details to README in certificate samples

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR adds more information to the README certificate samples to have an easier complete guide for SSL/TLS certificate configuration.

## Proposed Changes
  - Updated Readme of dotnet samples 84.bot-authentication-certificate and 85.bot-authentication-sni.

### CODE REVIEW

File: samples/csharp_dotnetcore/85.bot-authentication-sni/README.md [hunk 1]
```diff
@@ -2,7 +2,13 @@
 
 Bot Framework v4 bot authentication using Subject Name/Issuer
 
-This bot has been created using [Bot Framework](https://dev.botframework.com/), is shows how to use the bot authentication capabilities of Azure Bot Service. In this sample, we use a local or KeyVault certificate and the MSAL Subject Name/Issuer configuration to create the Bot Framework Authentication.
+This bot has been created using Bot Framework, is shows how to use the bot authentication capabilities of Azure Bot Service. In this sample, we use a local or KeyVault certificate and the MSAL Subject Name/Issuer configuration to create the Bot Framework Authentication.
```
Comment (andres-robinet-sw) [inline samples/csharp_dotnetcore/85.bot-authentication-sni/README.md]: ", it shows" instead of ", is shows"

**Comment by JhontSouth** [comment]
Include az login step

File: samples/csharp_dotnetcore/85.bot-authentication-sni/README.md (hunk 1 above)
Comment (JhontSouth) [reply samples/csharp_dotnetcore/85.bot-authentication-sni/README.md]: fixed

**Comment by sw-joelmut** [comment]
I created https://github.com/southworks/BotBuilder-Samples/pull/429 to improve the readme, and merge it to this pr once approved

For SNI it also applies the same principles

File: samples/csharp_dotnetcore/85.bot-authentication-sni/README.md [hunk 2]
```diff
@@ -2,7 +2,13 @@
 
 Bot Framework v4 bot authentication using Subject Name/Issuer
 
-This bot has been created using [Bot Framework](https://dev.botframework.com/), is shows how to use the bot authentication capabilities of Azure Bot Service. In this sample, we use a local or KeyVault certificate and the MSAL Subject Name/Issuer configuration to create the Bot Framework Authentication.
+This bot has been created using Bot Framework, it shows how to use the bot authentication capabilities of Azure Bot Service. In this sample, we use a local or KeyVault certificate and the MSAL Subject Name/Issuer configuration to create the Bot Framework Authentication.
+>NOTE: Microsoft's first-party resources are required to test this sample.
```
Comment (sw-joelmut) [inline samples/csharp_dotnetcore/85.bot-authentication-sni/README.md]: I will add this NOTE like this. More info in https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
> [!IMPORTANT]
> Microsoft's first-party resources are required to test this sample.

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: samples/csharp_dotnetcore/84.bot-authentication-certificate/README.md [hunk 3]
```diff
@@ -164,6 +164,11 @@ An SSL/TLS certificate is a digital object that allows systems to verify identit
 
       ![Certificate Reading](Images/KeyVault/CertificateReading.png)
```
Comment (andres-robinet-sw) [inline samples/csharp_dotnetcore/84.bot-authentication-certificate/README.md:160]: Case sensitivity of links is important if the repo is cloned in linux/unix systems. `Images` is not the same as `images`

**Review by andres-robinet-sw — CHANGES_REQUESTED** [review]

File: samples/csharp_dotnetcore/84.bot-authentication-certificate/README.md (hunk 3 above)
Comment (JhontSouth) [reply samples/csharp_dotnetcore/84.bot-authentication-certificate/README.md:160]: in this case the capital letter matches, in JS it doesn't

File: samples/csharp_dotnetcore/85.bot-authentication-sni/README.md (hunk 2 above)
Comment (JhontSouth) [reply samples/csharp_dotnetcore/85.bot-authentication-sni/README.md]: done

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
We should replace this https://github.com/southworks/BotBuilder-Samples/pull/421/files#diff-7dbeff6709649d9c6e9229f72bcdc4d92ab7e4b89670961266729f98010d7769R55-R58 with the CertificateServiceClientCredentialsFactory like the normal certificate sample, but sendX5c.

**Comment by JhontSouth** [comment]
> We should replace this https://github.com/southworks/BotBuilder-Samples/pull/421/files#diff-7dbeff6709649d9c6e9229f72bcdc4d92ab7e4b89670961266729f98010d7769R55-R58 with the CertificateServiceClientCredentialsFactory like the normal certificate sample, but sendX5c.

I guess that you are referring to this(the link doesn't work):
services.AddSingleton<ServiceClientCredentialsFactory>(
                new CertificateServiceClientCredentialsFactory(certificate, _configuration["MicrosoftAppId"], _configuration["MicrosoftAppTenantId"], null, null, true));

 
Instead of:
 
            // MSAL certificate auth.
            services.AddSingleton(
                serviceProvider => ConfidentialClientApplicationBuilder.Create(_configuration["MicrosoftAppId"])
                    .WithCertificate(certificate, sendX5C)
                    .Build());
 
            // MSAL credential factory: regardless of secret, cert or custom auth, need to add the line below to enable MSAL.
            services.AddSingleton<ServiceClientCredentialsFactory, MsalServiceClientCredentialsFactory>();

**Comment by sw-joelmut** [comment]
Yes, also the image in the readme should be updated.

**Comment by JhontSouth** [comment]
> services.AddSingleton(
> new CertificateServiceClientCredentialsFactory(certificate, _configuration["MicrosoftAppId"], _configuration["MicrosoftAppTenantId"], null, null, true));

solved

**Review by sw-joelmut — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /3995

**Pull request closed without merging**